#embed detection fails hard on older gcc

<!-- Please search existing issues to avoid creating duplicates. -->

#### Example of problem
since the addition of `#embed` detection in #4265 running configure on OpenBSD/sparc64 (which uses an oooldish gcc 8.4) blows hard:
```
-- Performing Test _TEST_SHARP_EMBED
CMake Error in /usr/obj/ports/proj-9.6.0/build-sparc64/CMakeFiles/CMakeScratch/TryCompile-Z6xBeR/CMakeLists.txt:
  Target "cmTC_b97b4" requires the language dialect "C23" .  But the current
  compiler "GNU" does not support this, or CMake does not know the flags to
  enable it.


CMake Error at /usr/local/share/cmake/Modules/Internal/CheckSourceCompiles.cmake:108 (try_compile):
  Failed to generate test project build system.
Call Stack (most recent call first):
  /usr/local/share/cmake/Modules/CheckCSourceCompiles.cmake:58 (cmake_check_source_compiles)
  CMakeLists.txt:365 (check_c_source_compiles)
  CMakeLists.txt:377 (is_sharp_embed_available)


-- Configuring incomplete, errors occurred!
```
#### Problem description
my understanding is that the `#embed` codepath is optional, and moreover maybe it shouldnt be tested if `EMBED_RESOURCE_FILES` is OFF (the default in my case ?)

#### Expected Output

configure should succeed.

#### Environment Information

- PROJ version (`proj`) 9.6.0
- Operation System Information OpenBSD/sparc64

#### Installation method

- conda, apt-get, from source, etc...

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -355,7 +355,8 @@ endif()
 
 function (is_sharp_embed_available res)
     if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.21 AND
-        ((CMAKE_C_COMPILER_ID STREQUAL "GNU") OR (CMAKE_C_COMPILER_ID STREQUAL "Clang")))
+        ((CMAKE_C_COMPILER_ID STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 15.0) OR
+         (CMAKE_C_COMPILER_ID STREQUAL "Clang" AND CMAKE_C_COMPILER_VERSION VERSION_GREATER_EQUAL 19.0)))
         # CMAKE_C_STANDARD=23 only supported since CMake 3.21
         set(TEST_SHARP_EMBED
           "static const unsigned char embedded[] = {\n#embed __FILE__\n};\nint main() { (void)embedded; return 0;}"

